src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -195,7 +195,7 @@ class TcgStatics {
    */
   static CardList selectEnergy(PokemonCardSet pcs, Type...types=C) {
     def ef = new SelectEnergy(pcs.cards, types)
-    ef.playerType = pcs.owner
+    ef.playerType = bg.currentTurn
```

Comment:
may I ask why this is needed? I would consider using `bg.currentThreadPlayerType` instead.

Reply:
The new method discardDefendingEnergyAfterDamage uses select energy, but builds the ui for the defending player rather than the attacking one. I discussed a few ways to do this with @flagrama, and we thought this was the best.